### PR TITLE
adding easyconfigs: HDF5-1.10.1-foss-2020a.eb and patches: HDF5-1.10.…

### PR DESCRIPTION
**Pull request by deniskristak**
…1-mpi_deprecated_methods.patch

upgrade for HDF5

### CODE REVIEW

File: easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    '3d564dca389eea5ce09e88476fa4c048ca0c79690859a133b09a46db58a60c88',
+]
+
+dependencies = [
+    ('OpenSSL', '1.1.1e'),
+    ('zlib', '1.2.11'),
+    ('Szip', '2.1.1'),
+    ('libxml2', '2.9.10'),
+]
+
```
Comment (akesandgren) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb:28]: We should probably turn on the testing, i.e.
runtest = "test"
on all new HDF5 easyconfigs.
Depends on having https://github.com/easybuilders/easybuild-easyblocks/pull/2250 though.

@boegel thoughts on doing this?

File: easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name = 'HDF5'
+version = '1.10.1'
```
Comment (Micket) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb:2]: Why this old bugfix version?
I know there are ABI differences between 1.12 and 1.10 series, but there is already a 1.10.6 version in 2020a.

Also, if this is truly needed, it should be moved down to gompi level.

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by deniskristak** [comment]
closing as it is not relevant anymore..found an alternative

File: easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-foss-2020a.eb:28]: @akesandgren Please open a dedicated issue for that, and we can do it in the most recent HDF5 easyconfigs?

**Pull request closed without merging**